pkg/reconcilers/deployment_mutator.go

```diff
@@ -88,17 +93,63 @@ func DeploymentImageMutator(desired, existing *k8sapps.Deployment) bool {
 	return update
 }
 
-func DeploymentVolumesMutator(desired, existing *k8sapps.Deployment) bool {
+func DeploymentImagePullPolicyMutator(desired, existing *k8sapps.Deployment) bool {
 	update := false
 
-	if !reflect.DeepEqual(existing.Spec.Template.Spec.Volumes, desired.Spec.Template.Spec.Volumes) {
-		existing.Spec.Template.Spec.Volumes = desired.Spec.Template.Spec.Volumes
+	if existing.Spec.Template.Spec.Containers[0].ImagePullPolicy != desired.Spec.Template.Spec.Containers[0].ImagePullPolicy {
+		existing.Spec.Template.Spec.Containers[0].ImagePullPolicy = desired.Spec.Template.Spec.Containers[0].ImagePullPolicy
+		update = true
+	}
+
+	return update
+}
+
+func DeploymentContainerArgsMutator(desired, existing *k8sapps.Deployment) bool {
+	update := false
+
+	existingArgs := existing.Spec.Template.Spec.Containers[0].DeepCopy().Args
```

Comment:
Should we nil check containers here to ensure we don't panic ?

Reply:
done